src/main/java/com/ldtteam/structurize/util/BlockUtils.java

```diff
@@ -220,6 +223,49 @@ else if (blockState.getBlock() == Blocks.BAMBOO_SAPLING)
         }
     }
 
+    /**
+     * For structure placement, check if two blocks are alike or if action has to be taken.
+     * @param blockState1 the first blockState.
+     * @param blockState2 the second blockState.
+     * @return true if nothing has to be done.
+     */
+    public static boolean areBlockStatesEqual(final BlockState blockState1, final BlockState blockState2)
+    {
+        if (blockState2 == null)
+        {
+            return true;
+        }
+
+        final Block structureBlock = blockState2.getBlock();
+        final Block worldBlock = blockState1.getBlock();
+
+        if (structureBlock == ModBlocks.blockSubstitution || blockState2.equals(blockState1))
+        {
+            return true;
+        }
+
+        if (structureBlock == ModBlocks.blockSolidSubstitution)
+        {
+            for (final Predicate<BlockState> predicate : BlockSolidSubstitution.NOT_SOLID)
+            {
+                if (predicate.test(blockState1))
+                {
+                    return true;
+                }
+            }
+        }
+
+        //list of things to only check block for.
+        //For the time being any flower pot is equal to each other.
+        if (structureBlock instanceof DoorBlock || structureBlock == Blocks.FLOWER_POT)
+        {
+            return structureBlock == worldBlock;
+        }
```

Comment:
If I'm not blind, this entire function is missing `structureBlock == worldBlock` check
Probably also remove this if?

Reply:
On purpose. However, I think we can remove the door part in 1.15 too. Although flowerpots we probably want to handle like this still.